File: final/exam/src/model/reason-detail.ts
import {Student} from "./student";
import {RewardAndPunishmentReason} from "./reward-and-punishment-reason";

export interface ReasonDetail {
  id? :number;
  dateOfOccurrence? :string;
  status? :number;
  student: Student;
  rewardAndPunishmentReason: RewardAndPunishmentReason;

}

File: final-be/src/main/java/com/example/model/ReasonDetail.java
package com.example.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ReasonDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private LocalDate dateOfOccurrence;

    private Integer status;

    @ManyToOne
    @JoinColumn(name = "reason_id")
    private RewardAndPunishmentReason reason;

    @ManyToOne
    @JoinColumn(name = "student_id")
    private Student student;
}
